### PR TITLE
[py] Subclass options classes from a common base class

### PR DESCRIPTION
**Pull request by lmtierney**
An ABC was used to ensure that `to_capabilities` was implemented. This method allows us to pass just the options class to `RemoteWebDriver` and not have to worry about capabilities.

---
This change is [Reviewable](https://reviewable.io/reviews/seleniumhq/selenium/6522)

### CODE REVIEW

**Comment by lmtierney** [comment]
@Dude-X

**Comment by lmtierney** [comment]
Safari options class will be added after I talk with @burg

**Comment by AutomatedTester** [comment]
from a quick look through this looks great

File: py/selenium/webdriver/chrome/options.py [hunk 1]
```diff
@@ -18,22 +18,21 @@
 import base64
 import os
 import platform
-import warnings
 
 from selenium.webdriver.common.desired_capabilities import DesiredCapabilities
+from ..common.options import ArgOptions
```
Comment (cgoldberg) [inline py/selenium/webdriver/chrome/options.py]: any reason for the relative import instead of absolute?

Comment (lmtierney) [reply py/selenium/webdriver/chrome/options.py]: Habit, I can change it to absolute

**Pull request merged**